### PR TITLE
Cleanup WatcherHandlerRegistration

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
replace https://github.com/istio/istio/pull/48469 @SpecialYang

File: pkg/config/mesh/networks_watcher.go [hunk 1]
```diff
@@ -34,10 +34,7 @@ type NetworksHolder interface {
 }
 
 // WatcherHandlerRegistration will be returned to caller to remove the handler later.
-type WatcherHandlerRegistration interface{}
-
-type watchHandler struct {
-	WatcherHandlerRegistration
+type WatcherHandlerRegistration struct {
 	handler func()
```
Comment (SpecialYang) [inline pkg/config/mesh/networks_watcher.go:38]: Here I'm little curious why using struct is better than using interface.

**Review by SpecialYang — APPROVED** [review]
Looks good. 👍🏻

File: pkg/config/mesh/networks_watcher.go [hunk 2]
```diff
@@ -46,17 +43,17 @@ type NetworksWatcher interface {
 	NetworksHolder
 
 	// AddNetworksHandler registers a callback handler for changes to the networks config.
-	AddNetworksHandler(func()) WatcherHandlerRegistration
+	AddNetworksHandler(func()) *WatcherHandlerRegistration
 
 	// DeleteNetworksHandler unregisters a callback handler when remote cluster is removed.
-	DeleteNetworksHandler(registration WatcherHandlerRegistration)
+	DeleteNetworksHandler(registration *WatcherHandlerRegistration)
```
Comment (howardjohn) [inline pkg/config/mesh/networks_watcher.go:49]: If I was designing it I would have 

```go
type Handler interface {
  Unsubscribe()
}
AddNetworksHandler(func()) Handler
```

Then they call
```go
handler := AddNetworksHandler(func(){})
handler.Unsubscribe()
```

Comment (howardjohn) [reply pkg/config/mesh/networks_watcher.go:49]: This works fine as  well though

**Review by howardjohn — APPROVED** [review]

File: pkg/config/mesh/networks_watcher.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pkg/config/mesh/networks_watcher.go:49]:
```
type Handler interface {
  Unsubscribe()
}
```

Sounds good

**Pull request merged**